extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -56,6 +56,13 @@ public HttpAuthenticator(IdentityProviderManager identityProviderManager,
         this.identityProviderManager = identityProviderManager;
         List<HttpAuthenticationMechanism> mechanisms = new ArrayList<>();
         for (HttpAuthenticationMechanism mechanism : httpAuthenticationMechanism) {
+            if (mechanism.getCredentialTypes().isEmpty()) {
+                // mechanism does not require any IdentityProvider
```

Comment:
@michalvavrik This is the minor change I was referring to, please add a debug log message

Reply:
Will do.